### PR TITLE
Move pose_graph_*d related files to internal/.

### PR DESCRIPTION
**Pull request by pifon2a**
[Internal headers RFC](https://github.com/googlecartographer/rfcs/blob/master/text/0003-internal-headers.md)

### CODE REVIEW

File: cartographer/mapping/internal/3d/rotation_parameterization.h [hunk 1]
```diff
@@ -14,8 +14,8 @@
  * limitations under the License.
  */
 
-#ifndef CARTOGRAPHER_MAPPING_3D_ROTATION_PARAMETERIZATION_H_
-#define CARTOGRAPHER_MAPPING_3D_ROTATION_PARAMETERIZATION_H_
+#ifndef CARTOGRAPHER_MAPPING_INTERNAL_3D_ROTATION_PARAMETERIZATION_H_
```
Comment (gaschler) [inline cartographer/mapping/internal/3d/rotation_parameterization.h]: mapping internal 3d or mapping 3d internal? (I don't care as long it's consistent.)

File: cartographer/mapping/internal/3d/pose_graph/optimization_problem_3d_test.cc [hunk 2]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-#include "cartographer/mapping/3d/pose_graph/optimization_problem_3d.h"
+#include "optimization_problem_3d.h"
```
Comment (gaschler) [inline cartographer/mapping/internal/3d/pose_graph/optimization_problem_3d_test.cc]: I haven't seen this for other tests (just make sure it's consistent)

Comment (pifon2a) [reply cartographer/mapping/internal/3d/pose_graph/optimization_problem_3d_test.cc]: ah, that's what CLion does sometimes. Thanks for noticing, I thought I cleaned up all of these.

File: cartographer/mapping/internal/3d/rotation_parameterization.h (hunk 1 above)
Comment (pifon2a) [reply cartographer/mapping/internal/3d/rotation_parameterization.h]: It is always MAPPING_INTERNAL

**Pull request merged**